src/logger.rs

```diff
@@ -16,11 +16,11 @@ extern "C" fn destructor(ctx: *mut c_void) {
     }
 }
 
-extern "C" fn logv(ctx: *mut c_void, log_level: InfoLogLevel, format: *const c_char, ap: VaList) {
+extern "C" fn logv(ctx: *mut c_void, log_level: InfoLogLevel, log: *const c_char) {
     unsafe {
         let logger = &*(ctx as *mut Box<dyn Logger>);
-        let format = CStr::from_ptr(format);
-        logger.logv(log_level, format.to_str().unwrap(), ap);
+        let log = CStr::from_ptr(log);
+        logger.logv(log_level, log.to_str().unwrap());
```

Comment:
Let's do `log.to_string_lossy()` here to avoid unexpected panic.

Reply:
Done